### PR TITLE
Fix ga tag pattern and add tests

### PR DESCRIPTION
**Pull request by gousteris**
This PR:
* fixes GA tag pattern
goreleaser runs on an ubuntu VM managed by gh actions. ubuntu doesn't
understand `\d` for digits. Instead, using the equivalent pattern
`[0-9]` that works fine on ubuntu:

```
$ version=0.0.0
$ pattern='^\d+\.\d+\.\d+$'
$ if [[ $version =~ $pattern ]]; then echo match; fi
$ pattern='^[0-9]+\.[0-9]+\.[0-9]+$'
$ if [[ $version =~ $pattern ]]; then echo match; fi
match
```
* adds gh action step to test push_pkg_to_cloudsmith.sh

### CODE REVIEW

File: resources/scripts/push_pkg_to_cloudsmith.sh [hunk 1]
```diff
@@ -40,7 +40,7 @@ else
     version=$PKG_VERSION
 fi
 
-GA_VERSION_PATTERN='^\d+\.\d+\.\d+$'
+GA_VERSION_PATTERN='^[0-9]+\.[0-9]+\.[0-9]+$'
```
Comment (ivotron) [inline resources/scripts/push_pkg_to_cloudsmith.sh:43]: would help if the commit message explains why this change is needed

**Review by ivotron — CHANGES_REQUESTED** [review]

**Review by ivotron — APPROVED** [review]

**Comment by peczenyj** [comment]
It os already done but you can also use posix classes like `[[:digit:]]`, if you think it is more readable
